core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -58,57 +60,60 @@ final case class InvokerInfo()
  * Note: An Invoker that never sends an initial Ping will not be considered
  * by the InvokerPool and thus might not be caught by monitoring.
  */
-class InvokerPool(kv: KeyValueStore, invokerDownCallback: String => Unit) extends Actor {
+class InvokerPool(
+    childFactory: (ActorRefFactory, String) => ActorRef,
+    kv: KeyValueStore,
+    invokerDownCallback: String => Unit) extends Actor {
+
     implicit val transid = TransactionId.invokerHealth
     val logging = new AkkaLogging(context.system.log)
     implicit val timeout = Timeout(5.seconds)
     implicit val ec = context.dispatcher
 
-    /** Collects the status of all invokers */
-    def collectStatus = {
-        val list = context.children.map { invoker =>
-            val name = invoker.path.name
-            invoker.ask(GetStatus).mapTo[InvokerState].map(name -> _)
-        }
-        Future.sequence(list).map(_.toMap)
-    }
+    val invokers = mutable.HashMap.empty[String, ActorRef]
+    val invokerStatus = mutable.HashMap.empty[String, InvokerState]
```

Comment:
Making fields in Actors private is somewhat redundant as there is no way to get a reference to the object itself. You only talk to the Actor via `ActorRef`